File: Data Structures & Algorithms/duplicate-integer/submission-5.rs
use std::collections::HashMap;
impl Solution {
    pub fn has_duplicate(nums: Vec<i32>) -> bool {
        let mut hmap = HashMap::new();
        for i in nums {
            if hmap.contains_key(&i) {
                return true;
            } else {
                hmap.insert(i, 1);
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-5.py
class Solution:
    
    def findDuplicate(self, nums: List[int]) -> int:
        f = nums[nums[0]]
        s = nums[0]

        while f != s:
            f = nums[nums[f]]
            s = nums[s]
        
        new = 0 
        while new != f:
            f = nums[f]
            new = nums[new]

        return new

            